nixos/modules/tasks/network-interfaces.nix

```diff
@@ -408,6 +408,9 @@ in
         (this derives it from the machine-id that systemd generates) or
 
         <literal>head -c4 /dev/urandom | od -A none -t x4</literal>
+
+        One use case is for ZFS to ensure that a pool isn't imported 
+        accidentally on a wrong machine. 
```

Comment:
Is there any other usecase than zfs? I recall having talked about this during the [networkd hackathon](https://discourse.nixos.org/t/networkd-sprint-2019-11-23-24-in-munich/4578/29) while revisiting the options inside `networking`.

Back then, we wanted to investigate whether it's used elsewhere, and spoke about moving it to the zfs module, which would make this much clearer.

Reply:
As much as I'd like for host id to be a part of zfs for clarity as an end user, I suspect that it belongs in networking because there is an actual command called `hostid` in coreutils which indicates that it belongs outside of zfs. 

That said, upon `rg` and reading files, I found 3 instances of `hostId` which do not relate to zfs 

```
nixos/tests/bcachefs.nix:7:    
nixos/tests/overlayfs.nix:7:    
nixos/modules/system/boot/stage-1-init.sh:195:
```

Of these, 3th is half related to zfs but also mentions "spl". What's that I wonder?

Reply:
SPL is the solaris porting layer. This was zfs abstraction layer in the past but now has been integrated into zfs itself. I suspect for the other tests in `bcachefs` and `overlayfs` someone just copy'n'pasted the host id from zfs.